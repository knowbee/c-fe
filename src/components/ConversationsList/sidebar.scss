.sidebar {
  position: fixed;
  width: 345px;
  border-right: 1px solid #ddd;
  margin-top: 3.5rem;
  padding-left: 0.8rem;
  overflow-y: scroll;
  overflow-x: hidden;
  top: 0;
  bottom: 0;
  &::-webkit-scrollbar {
    width: 3px; /* width of the entire scrollbar */
  }

  &::-webkit-scrollbar-thumb {
    background-color: #2aa2fa; /* color of the scroll thumb */
    border-radius: 20px; /* roundness of the scroll thumb */
  }
  .sidebar-header {
    border-bottom: 1px solid #ddd;
    margin-left: -1rem;
    margin-right: -0.1rem;
    margin-bottom: 1rem;
    padding: 0.8rem;
    input {
      background-color: rgba(
        250.25195717811584,
        250.25195717811584,
        250.25195717811584,
        1
      );
      opacity: 1;
      width: 20rem;
      height: 3rem;
      border: 1px solid #ddd;
    }
    button {
      background-color: #fff;
      border: none;

      left: 19rem;
      top: 7rem;
      cursor: pointer;
    }
    .search-icon {
      position: relative;
      left: 18rem;
      top: -2.2rem;
    }

    .active {
      display: flex;
      justify-content: space-between;
      p {
        margin: 0rem;
        color: #2aa2fa;
      }

      svg {
        margin-top: 0.5rem;
      }
    }
  }
  .sidebar-body {
    margin-left: -0.8rem;
    
    .active-chat{
      background-color: #EDEDED;
    }
    .chat{
      display: flex;
      width: 100% !important;
      padding: 1px;
      flex-direction: row;
      justify-content: space-between;
      align-items: center;

    }
    ul {
      margin: 0rem;
      padding: 0rem;

      #link {
        list-style: none;
        &:target {
          // background-color: #bbb;
        }
        cursor: pointer;
        a {
          text-decoration: none;
          color: inherit;
          display: flex;
          padding: 1.5rem 1rem;
          &:hover {
            background-color: #ddd;
          }

          img {
            width: 50px;
            height: 50px;
            border-radius: 50%;
            margin-right: 1rem;
          }

          div {
            span {
              display: flex;
              justify-content: space-between;
              h4 {
                margin-bottom: 0.2rem;
                margin-top: 0rem;
              }
            }

            p {
              margin: 0rem;
            }
          }
        }
      }
    }
  }
}
