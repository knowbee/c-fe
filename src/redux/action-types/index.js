export * from "./registerUserTypes";
export * from "./loginTypes";
export * from "./currentUserTypes";
export * from "./usersTypes";
