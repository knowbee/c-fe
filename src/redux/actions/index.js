export * from "./registerActions";
export * from "./loginActions";
export * from "./currentUserActions";
