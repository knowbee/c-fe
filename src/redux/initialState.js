module.exports = {
  currentUser: {
    message: "",
    isLoggedIn: localStorage.getItem("token") !== null,
    profile: {},
  },
  register: {
    submitting: false,
    message: "",
    email: "",
    name: "",
    password: "",
    errors: {},
  },
  users: {
    fetching: false,
    chats: [],
    errors: {},
  },
  login: {
    submitting: false,
    credentials: {
      email: "",
      password: "",
    },
  },
};
