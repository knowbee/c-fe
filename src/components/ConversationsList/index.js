/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState, useRef, useEffect } from "react";
import "./sidebar.scss";
import { connect } from "react-redux";
import Moment from "react-moment";

import {
  createReceiverId,
  createReceiverName,
  fetchMessages,
} from "../../redux/actions";
function Sidebar({ chats, saveReceiverId, saveReceiverName, fetchMessages }) {
  const [selectedUser, setSelectedUser] = useState();

  const chatsEndRef = useRef(null);

  useEffect(() => {
    scrollToBottom();
  }, [selectedUser]);

  const scrollToBottom = () => {};
  const selectChat = (chat) => {
    setSelectedUser(chat);
  };
  return (
    <div className="sidebar">
      <div className="sidebar-header">
        {" "}
        <div className="active">
          {" "}
          <p>Messages</p>
        </div>
      </div>
      <div className="sidebar-body">
        <ul>
          {chats.map((item) => (
            <a
              key={item.id}
              id="link"
              onClick={() => {
                selectChat(item);
                saveReceiverId(item.id);
                saveReceiverName(item.name);
                fetchMessages(item.id);
              }}
            >
              <div
                className={
                  selectedUser && selectedUser.id === item.id
                    ? "active-chat"
                    : ""
                }
              >
                <div className="chat">
                  <h4>{item.name}</h4>
                  <div>
                    <Moment format="h:mm:ss a">{item.joined_at}</Moment>
                  </div>
                </div>
              </div>
            </a>
          ))}
        </ul>
      </div>
      <div ref={chatsEndRef} />
    </div>
  );
}

export const mapDispatchToProps = (dispatch) => ({
  saveReceiverId: (id) => dispatch(createReceiverId(id)),
  saveReceiverName: (name) => dispatch(createReceiverName(name)),
  fetchMessages: (id) => dispatch(fetchMessages(id)),
});

export default connect(null, mapDispatchToProps)(Sidebar);
