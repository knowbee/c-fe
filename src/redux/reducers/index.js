import { combineReducers } from "redux";
import register from "./registerReducer";
import login from "./loginReducer";
import currentUser from "./currentUserReducer";
import users from "./usersReducer";

const reducer = combineReducers({
  register,
  login,
  currentUser,
  users,
});
export default reducer;
