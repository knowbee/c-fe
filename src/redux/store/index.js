import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import initialState from "../initialState";
import reducers from "../reducers/index";

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
