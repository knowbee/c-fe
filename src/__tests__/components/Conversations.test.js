import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";

import "./index.css";
import rootReducer from "../../redux/reducers";
import ConversationsList from "../../components/ConversationsList";

const store = createStore(rootReducer);

test("ConversationsList component loads with default state", () => {
  const chats = [];
  const component = renderer.create(
    <Provider store={store}>
      <ConversationsList chats={chats} />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
