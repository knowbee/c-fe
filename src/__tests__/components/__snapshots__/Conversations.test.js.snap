// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationsList component loads with default state 1`] = `
<div
  className="sidebar"
>
  <div
    className="sidebar-header"
  >
     
    <div
      className="active"
    >
       
      <p>
        Messages
      </p>
    </div>
  </div>
  <div
    className="sidebar-body"
  >
    <ul />
  </div>
  <div />
</div>
`;
